addon/components/file-dropzone.ts

```diff
@@ -0,0 +1,354 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { getOwner } from '@ember/application';
+import DataTransferWrapper, {
+  FileUploadDragEvent,
+} from '../system/data-transfer-wrapper';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import Queue from '../queue';
+import UploadFile, { FileSource } from 'ember-file-upload/upload-file';
+import FileQueueService, { DEFAULT_QUEUE } from '../services/file-queue';
+import { modifier } from 'ember-modifier';
+
+interface FileDropzoneArgs {
+  queue?: Queue;
+
+  /**
+   * Whether users can upload content from websites by dragging images from
+   * another webpage and dropping it into your app. The default is `false`
+   * to prevent cross-site scripting issues.
+   * */
+  allowUploadsFromWebsites?: boolean;
+
+  /**
+   * This is the type of cursor that should
+   * be shown when a drag event happens.
+   *
+   * Corresponds to `DataTransfer.dropEffect`.
+   * (https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect)
+   */
+  cursor?: 'link' | 'none' | 'copy' | 'move';
+
+  // actions
+  filter?: (file: UploadFile) => boolean;
+
+  /**
+   * Called when files have entered the dropzone.
+   */
+  filesEnter?: (
```

Comment:
That name seems inconsistent to me with the rest, it may be:

- `enterFile(s)` - apply business logic to mutate the data when entering the zone
- `filesEntered` - only a notifier, that some files have entered the zone (but no chance of altering them)